packages/react-ui-core/src/DatePicker/Calendar.js

```diff
@@ -43,6 +48,34 @@ export default class Calendar extends PureComponent {
     }
   }
 
+  get daysOfWeek() {
+    const daysOfWeek = []
+    const monthStart = startOfMonth(this.state.date)
+    const dow = getDay(startOfMonth(this.state.date))
+    const firstDate = subDays(monthStart, dow)
+
+    for (let i = 0; i <= 6; i += 1) {
+      daysOfWeek.push(<div key={`dow-${i}`}>{format(addDays(firstDate, i), 'EEEEEE')}</div>)
+    }
+
+    return daysOfWeek
+  }
+
+  get daysOfMonth() {
+    const days = []
+
+    // NOTE: consitently create 5 rows so the calendar pads shorter months
+    for (let i = 0; i <= 5; i += 1) {
+      days.push(
+        <div key={`date-row-${i}`}>
```

Comment:
I don't think there is any need to make the key unique, why not just `key={i}`

Reply:
Oh, maybe this array gets concatenated with this other, so you might need that afterall. If so please ignore.